src/Interpreters/ActionsDAG.cpp

```diff
@@ -796,11 +795,18 @@ ActionsDAGPtr ActionsDAG::foldActionsByProjection(const std::unordered_map<const
 
                     if (!node)
                     {
-                        bool should_rename = !rename.empty() && new_input->result_name != rename;
-                        const auto & input_name = should_rename ? rename : new_input->result_name;
-                        node = &dag->addInput(input_name, new_input->result_type);
-                        if (should_rename)
-                            node = &dag->addAlias(*node, new_input->result_name);
+                        auto & mapped_input = inputs_mapping[rename];
+
+                        if (!mapped_input)
```

Comment:
It is possible to have a few aliases on the same column. We may want to replace all the aliases, in this case they should have a single input as a child.